graylog2-web-interface/src/components/cluster-configuration/data-nodes/DataNodesColumnConfiguration.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+
+import type { ColumnRenderers, ColumnSchema } from 'components/common/EntityDataTable';
+import { Link } from 'components/common/router';
+import DataNodeStatusCell from 'components/datanode/DataNodeList/DataNodeStatusCell';
+import type { DataNode } from 'components/datanode/Types';
+import Routes from 'routing/Routes';
+
+import type { ClusterDataNode } from './useClusterDataNodes';
+import CpuMetricsCell from './cells/CpuMetricsCell';
+import IndexingMetricsCell from './cells/IndexingMetricsCell';
+
+import { RoleLabel, SecondaryText } from '../shared-components/NodeMetricsLayout';
+import SizeAndRatioMetric from '../shared-components/SizeAndRatioMetric';
+
+export const DEFAULT_VISIBLE_COLUMNS = [
+  'hostname',
+  'opensearch_roles',
+  'datanode_version',
+  'datanode_status',
+  'cpu',
+  'memory',
+  'jvm',
+  'indexing',
+  'storage',
+] as const;
+
+const JVM_WARNING_THRESHOLD = 0.95;
+const MEMORY_WARNING_THRESHOLD = 0.95;
+const STORAGE_WARNING_THRESHOLD = 0.7;
+const STORAGE_DANGER_THRESHOLD = 0.8;
+
+export const createColumnDefinitions = (): Array<ColumnSchema> => [
+  { id: 'hostname', title: 'Node', sortable: true },
+  { id: 'opensearch_roles', title: 'Role', sortable: true },
+  { id: 'datanode_version', title: 'Version', sortable: true },
+  { id: 'datanode_status', title: 'State', sortable: true },
+  { id: 'cpu', title: 'CPU', sortable: false, isDerived: true },
+  { id: 'memory', title: 'Memory', sortable: false, isDerived: true },
+  { id: 'jvm', title: 'JVM', sortable: false, isDerived: true },
+  { id: 'indexing', title: 'Indexing', sortable: false, isDerived: true },
+  { id: 'storage', title: 'Storage', sortable: false, isDerived: true },
+];
```

Comment:
What was the reason to not make use of the `attributes` from the backend response here?